synapse/config/server.py

```diff
@@ -542,6 +542,14 @@ class LimitRemoteRoomsConfig:
             users_new_default_push_rules
         )  # type: set
 
+        # Whitelist of domain names that given next_link parameters must have
+        next_link_domain_whitelist = config.get("next_link_domain_whitelist")
+        if not isinstance(next_link_domain_whitelist, list):
```

Comment:
Surely this will fail if the setting isn't provided (which isn't what the doc in the config says)? Given `d.get` returns `None` by default if the key wasn't found.